### DISCUSSION TITLE
Standalone and embeddable version of GDScript (2.0)?

### DISCUSSION THREAD

**Discussion by bandaloo** [godotengine/godot-proposals#6199] [Scripting]
I know it's been said before that GDScript is designed around Godot and tightly integrated into the engine. Additionally, I'm aware this has been [discussed](https://github.com/godotengine/godot/issues/17846) in 2017 and was subsequently closed. Because it's been a hot minute, and GDScript 2.0 looks *really* fantastic, I was wondering if anybody has thought about this again.

I won't enumerate all the cool things in GDScript 2.0, but with everything combined it now matches Lua in terms of feature parity IMO, and then goes beyond by optimizing common uses cases in games, even if straight-up speed doesn't match LuaJIT yet. There aren't many gradually typed scripting languages that *actually* make use of the type information to increase performance (Typescript doesn't, Python doesn't, even Roblox's Luau doesn't) so as a standalone piece of tech, GDScript is pretty exciting. It would be amazing if it were as easy to embed (or just install) as LuaJIT. It might encourage even more PL experts to contribute to GDScript itself while the Godot Engine reaps the benefits. I honestly think GDScript has broader applications than the creators itself give it credit for.

If this is not worth discussing again, or if it's already officially on the roadmap, feel free to close this thread. Just wanted to say I'm in awe by all the amazing work that's been done on GDScript and is slated for Godot 4.0 :)

**Comment by Calinou** [comment]
GDScript 2.0 is still designed for tight engine integration, using Godot's custom datatypes. I don't think it would fare well as a standalone language, let alone one that is embedded in other programs with different dataypes.

There's also the issue that by making the language standalone, we'd have to cater to the needs of projects not related to Godot at all. This broadens the scope for a programming language a lot, and it's easy to end up with scope creep here.

**Comment by theraot** [comment]
You can use Godot to run a GDScript file from command lines. However, that won't make it easy to call into whatever application you are trying to add scripting to.

Instead, you could have a Godot game where you add whatever you need to make available to scripting, and then and call it with command line arguments to pass the script. I can imagine your application accepting connections from the loopback address over a particular port, allowing the Godot game to interface with it over it. And, of course, another command line argument would be port it needs to use. At least that should work on Desktop. Bonus points: if the script crashes or hangs, it does not have to affect your application.

By the way, you might be interested in https://github.com/godotengine/godot-proposals/issues/4773
